src/WebJobs.Script.WebHost/FunctionsSyncService.cs

```diff
@@ -43,7 +45,8 @@ internal bool ShouldSyncTriggers
             get
             {
                 return _primaryHostStateProvider.IsPrimary &&
-                       (_scriptHostManager.State == ScriptHostState.Running);
+                       (_scriptHostManager.State == ScriptHostState.Running) &&
+                       !_environment.IsInValidationMode();
```

Comment:
This looks like it will impact non-flex skus. We should ensure this doesn't happen.